src/plugins/vis_type_vega/public/vega_visualization.ts

```diff
@@ -116,18 +116,17 @@ export const createVegaVisualization = ({ getServiceSettings }) =>
         };
 
         if (vegaParser.useMap) {
-          const services = { toastService: getNotifications().toasts };
           const { VegaMapView } = await import('./vega_view/vega_map_view');
-          this._vegaView = new VegaMapView(vegaViewParams, services);
+          this.vegaView = new VegaMapView(vegaViewParams);
         } else {
-          const { VegaView } = await import('./vega_view/vega_view');
-          this._vegaView = new VegaView(vegaViewParams);
+          const { VegaView: VegaViewClass } = await import('./vega_view/vega_view');
+          this.vegaView = new VegaViewClass(vegaViewParams);
         }
-        await this._vegaView.init();
+        await this.vegaView?.init();
```

Comment:
I think here should be `await this.vegaView!.init();` I prefer to see error if for some reasons `this.vegaView` is null. But not sure that this case is possible